programs/zstdcli_trace.c

```diff
@@ -82,7 +82,9 @@ static void TRACE_log(char const* method, PTime duration, ZSTD_Trace const* trac
     int level = 0;
     int workers = 0;
     double const ratio = (double)trace->uncompressedSize / (double)trace->compressedSize;
-    double const speed = ((double)trace->uncompressedSize * 1000) / (double)duration;
+    const double speed = duration != 0
+        ? ((double)trace->uncompressedSize * 1000) / (double)duration
+        : 0.0;
```

Comment:
Zero speed? Shall not it be INFINITY either?..

However, since duration is a precise time time counter (and it's nanosecond precise), I don't think the duration as diff between 2 measurements could be ever 0, because would mean few CPU instructions.
But to be sure (and safe in the future), I'd suggest simply something below its min-value (e. g. small fraction like `.1` or `1e-4`):
```diff
- double const speed = ((double)trace->uncompressedSize * 1000) / (double)duration;
+ double const speed = ((double)trace->uncompressedSize * 1000) / (duration ? (double)duration : .1);
```